src/main/java/marquez/service/models/JobId.java

```diff
@@ -0,0 +1,13 @@
+package marquez.service.models;
+
+import lombok.NonNull;
+import lombok.Value;
+import marquez.common.models.JobName;
+import marquez.common.models.NamespaceName;
+
+@Value
+public class JobId {
+
+  @NonNull NamespaceName namespace;
+  @NonNull JobName name;
```

Comment:
What are your thoughts on having the underlying value of the ID be a _string_? That is, we'd define a factory class to create instances of `JobId `:

```java
NamespaceName NamespaceName = NamespaceName.of("my_namespace");
JobName jobName = JobName.of("my_job");

JobId id = JobId.of(NamespaceName, JobName)`
log.info(id.getValue())
```
```
>> my_namespace:my_job
```

Reply:
As the url of a Job resource is ```/namespaces/{namespceName}/jobs/{name}``` it makes sense to me that the ID calls those out its components explicitly. That being said we could have something more compact. I'd use a separator (like ```:```) only if we have a non ambiguous way of parsing it back (for example namespace and name don't contain ```:```) . If we wanted this, I'd keep JobId as is but would add converters to serialize it as a string using the JsonSerializer annotation.

Reply:
I think this makes a lot of sense. A job or dataset is unique within a namespace and the composite of the two (`dataset|job`+ `namespace`) should make up the ID itself. By introducing a separator, we are present with other concerns. So, making each component explicit avoids us having to define constraints on what valid characters can be present in a name.